### PR TITLE
Fix Doxygen warnings

### PR DESCRIPTION
**Pull request by HalynaDumych**
Add API doc to omit Doxygen errors.

Relates-To: OLPESGE-731
Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 1]
```diff
@@ -286,10 +286,24 @@ class AUTHENTICATION_API AuthenticationClient {
...
+   * The `AuthenticationClient` instance is movable but not copiable.
+   */
   AuthenticationClient(const AuthenticationClient&) = delete;
+
+  /**
+   * @brief A copy assignment operator.
+   *
+   * The `AuthenticationClient` instance is movable but not copiable.
+   */
   AuthenticationClient& operator=(const AuthenticationClient&) = delete;
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h]: Changed

File: olp-cpp-sdk-core/include/olp/core/http/NetworkRequest.h [hunk 2]
```diff
@@ -46,7 +44,7 @@ class CORE_API NetworkRequest final {
     PUT = 3,     ///< The PUT method (RFC2616 section-9.6).
     DEL = 4,     ///< The DELETE method (RFC2616 section-9.7).
     PATCH = 5,   ///< The PATCH method (RFC2068 section-19.6.1.1).
-    OPTIONS = 6, ///< The OPTIONS method (RFC2616 section-9.2).
+    OPTIONS = 6,  ///< The OPTIONS method (RFC2616 section-9.2).
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-core/include/olp/core/http/NetworkRequest.h]: Changed

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 3]
```diff
@@ -286,10 +286,20 @@ class AUTHENTICATION_API AuthenticationClient {
...
 
-  // Non-copyable but movable
+  /// A copy constructor.
   AuthenticationClient(const AuthenticationClient&) = delete;
+
+  /**
+   * @brief A copy assignment operator.
+   *
+   * The `AuthenticationClient` instance is movable but not copiable.
+   */
```
Comment (andescu) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:296]:
```suggestion
  /// A copy assignment operator.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h [hunk 4]
```diff
@@ -65,10 +65,32 @@ class TokenProvider {
       : impl_(std::make_shared<TokenProviderImpl>(
             std::move(settings), std::chrono::seconds(MinimumValidity))) {}
 
-  // Default copyable, default movable.
+  /**
+   * @brief A default copy constructor.
+   *
+   * @param other The `TokenProvider` instance.
+   */
```
Comment (andescu) [inline olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h:72]:
```suggestion
  /// A default copy constructor.
```

Same below and for all other classes.
The following constructors and operators I would only comment with a short one liner:
- default constructor
- destructor (if even needed as it is pretty obvious)
- default copy constructor
- default move constructor
- default copy assignment operator
- default move assignment operator

**Review by andescu — APPROVED** [review]

**Review by jmoro3 — APPROVED** [review]

**Pull request merged**